Warning during compilation

Hello,

I compiled using Visual Studio 2019 and found this warning

1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\xtgmath.h(24,70): warning C4244: 'argument': conversion from 'double' to 'T', possible loss of data
1> with
1> [
1> T=uint32_t
1> ] (compiling source file ..\src\networkmessage.cpp)

--- a/src/networkmessage.cpp
+++ b/src/networkmessage.cpp
@@ -64,7 +64,7 @@ void NetworkMessage::addString(const std::string& value)
 void NetworkMessage::addDouble(double value, uint8_t precision/* = 2*/)
 {
 	addByte(precision);
-	add<uint32_t>((value * std::pow(static_cast<float>(10), precision)) + std::numeric_limits<int32_t>::max());
+	add<uint32_t>(static_cast<uint32_t>((value * std::pow(static_cast<float>(10), precision)) + std::numeric_limits<int32_t>::max()));
 }
 
 void NetworkMessage::addBytes(const char* bytes, size_t size)

